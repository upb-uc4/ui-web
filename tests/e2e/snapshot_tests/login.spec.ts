describe.skip("Login page snapshot testing", () => {
    it("toMatchImageSnapshot - whole page", () => {
        cy.visit("/").then(() => {
            cy.document().toMatchImageSnapshot();
        });
    });
});
