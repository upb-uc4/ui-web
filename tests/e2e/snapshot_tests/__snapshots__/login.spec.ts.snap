exports[`Login page snapshot testing > toMatchImageSnapshot - whole page #0`] = `
<div id="app" data-v-app="">
  <div id="root" class="">
    <div>
      <header
        class="flex items-center justify-between px-4 bg-gray-800 lg:px-8 md:h-20 min-h-20 hidden md:flex"
      >
        <div class="flex items-center w-full">
          <div class="flex items-center pr-8 mr-8 border-r-4 border-gray-100">
            <button
              class="flex text-3xl font-semibold tracking-wider text-gray-100 outline-none"
            >
              UC4
            </button>
          </div>
          <div class="flex w-full">
            <nav class="items-center justify-between w-full md:flex hidden">
              <ul
                class="justify-between pt-2 font-semibold tracking-wider text-gray-100 md:flex"
              >
                <li>
                  <a
                    href="/login"
                    id="nav_desktop_guest_login"
                    class="navbar-menu"
                    >Login</a
                  >
                </li>
                <li>
                  <a href="/about" id="routeAbout" class="navbar-menu">About</a>
                </li>
              </ul>
              <div class="pb-3"></div>
            </nav>
          </div>
          <button
            type="button"
            class="block text-gray-100 md:hidden hover:text-gray-500 focus:outline-none"
          >
            <svg class="w-8 h-8 fill-current" viewBox="0 0 24 24">
              <!--v-if-->
              <path
                fill-rule="evenodd"
                d="M4 5h16a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2z"
              ></path>
            </svg>
          </button>
        </div>
      </header>
      <header
        class="flex flex-col w-full px-4 bg-gray-800 min-h-16 flex md:hidden"
      >
        <div class="flex justify-between p-2">
          <div class="flex pr-8 mr-8 border-gray-100">
            <button
              class="text-3xl font-semibold tracking-wider text-gray-100 outline-none"
            >
              UC4
            </button>
          </div>
          <button
            id="nav_mobile_toggle_menu"
            type="button"
            class="block text-gray-100 md:hidden hover:text-gray-500 focus:outline-none"
          >
            <svg class="w-8 h-8 fill-current" viewBox="0 0 24 24">
              <!--v-if-->
              <path
                fill-rule="evenodd"
                d="M4 5h16a1 1 0 0 1 0 2H4a1 1 0 1 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2zm0 6h16a1 1 0 0 1 0 2H4a1 1 0 0 1 0-2z"
              ></path>
            </svg>
          </button>
        </div>
        <nav class="hidden">
          <ul class="pb-4 font-semibold tracking-wider text-gray-100">
            <li>
              <a
                href="/login"
                id="nav_mobile_guest_login"
                class="rounded-lg hover:bg-blue-800 mobile-navbar-menu"
                >Login</a
              >
            </li>
            <li>
              <a
                href="/about"
                id="nav_mobile_common_about"
                class="hover:bg-blue-800 rounded-lg mobile-navbar-menu"
                >About</a
              >
            </li>
          </ul>
        </nav>
      </header>
    </div>
    <div class="container mx-auto px-4">
      <div>
        <div
          class="container flex flex-col h-auto max-w-full md:mt-32 lg:flex-row lg:items-center"
        >
          <form
            method="POST"
            action=""
            class="flex flex-col items-center w-full mx-auto xl:w-3/4"
          >
            <h1
              class="mt-2 mb-10 text-4xl font-bold text-center text-gray-900 lg:text-5xl"
            >
              Login to Your Account
            </h1>
            <div
              class="items-center justify-center flex-auto w-full mx-4 lg:w-3/5"
            >
              <div class="mb-6 text-center">
                <i
                  class="absolute m-3 mt-4 ml-4 text-gray-500 fas fa-envelope"
                ></i
                ><input
                  id="email"
                  class="pl-10 font-semibold lg:w-3/4 form-input input-text"
                  type="text"
                  placeholder="Email"
                />
              </div>
              <div class="mb-6 text-center">
                <i class="absolute m-3 mt-4 ml-4 text-gray-500 fas fa-lock"></i
                ><input
                  id="password"
                  type="password"
                  class="pl-10 font-semibold lg:w-3/4 form-input input-text"
                  placeholder="Password"
                /><button
                  id="togglePassword"
                  type="button"
                  tabindex="-1"
                  class="absolute mt-1 ml-3 text-lg text-gray-500 hover:text-gray-600 focus:outline-none"
                >
                  <i class="fa-eye absolute mt-3 ml-1 mr-1 fas"></i></button
                ><!--v-if-->
              </div>
              <div class="w-full text-center lg:text-left lg:pl-16 lg:ml-3">
                <label class="block font-semibold text-gray-500"
                  ><input
                    id="rememberMe"
                    class="mr-2 text-blue-500 form-checkbox hover:bg-blue-600"
                    type="checkbox"
                    checked=""
                  /><span class="text-sm">Remember me</span></label
                ><a
                  id="forgotPassword"
                  class="inline-block mt-2 text-sm font-semibold navigation-link"
                  href="#"
                >
                  Forgot Password?
                </a>
              </div>
              <div class="justify-center mt-10 mb-6 text-center">
                <button
                  id="login"
                  type="submit"
                  disabled=""
                  class="inline-block w-2/5 sm:w-2/5 md:w-2/5 lg:w-2/4 center btn btn-blue-primary"
                >
                  Login
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
      <div id="modal-wrapper"></div>
    </div>
    <div
      class="fixed w-full h-full top-0 left-0 flex items-center justify-center"
      data-v-1875efe0=""
      style="display: none;"
    >
      <div
        class="absolute w-full h-full bg-gray-900 opacity-25"
        data-v-1875efe0=""
      ></div>
      <div
        class="bg-white w-full -mt-32 md:max-w-md lg:max-w-lg xl:max-w-xl mx-auto rounded-lg shadow-xl z-50 overflow-y-auto"
        data-v-1875efe0=""
      >
        <div class="py-8 px-10" data-v-1875efe0="">
          <div class="flex justify-between items-center" data-v-1875efe0="">
            <p class="text-2xl text-gray-900">Login</p>
            <button
              id="baseModalCancel"
              class="cursor-pointer z-50 focus:outline-none"
              data-v-1875efe0=""
            >
              <svg
                class="fill-current text-gray-600 hover:text-gray-700"
                xmlns="http://www.w3.org/2000/svg"
                width="28"
                height="28"
                viewBox="0 0 18 18"
                data-v-1875efe0=""
              >
                <path
                  d="M14.53 4.53l-1.06-1.06L9 7.94 4.53 3.47 3.47 4.53 7.94 9l-4.47 4.47 1.06 1.06L9 10.06l4.47 4.47 1.06-1.06L10.06 9z"
                  data-v-1875efe0=""
                ></path>
              </svg>
            </button>
          </div>
          <div
            class="flex mt-4 mr-8 text-lg text-gray-600 leading-relaxed tracking-wide"
            data-v-1875efe0=""
          >
            <div class="flex flex-col" data-v-1875efe0="">
              <p>Please enter your authentication credentials.</p>
              <div class="mb-4 mt-4" data-v-1875efe0="">
                <i
                  class="mt-4 m-3 ml-4 fas fa-envelope absolute text-gray-500"
                ></i
                ><input
                  id="loginModalEmail"
                  class="lg:w-5/6 font-semibold pl-10 form-input input-text"
                  type="text"
                  placeholder="Email"
                  data-v-1875efe0=""
                />
              </div>
              <div class="mb-2" data-v-1875efe0="">
                <i class="mt-4 m-3 ml-4 fas fa-lock absolute text-gray-500"></i
                ><input
                  id="loginModalPassword"
                  type="password"
                  class="lg:w-5/6 font-semibold pl-10 form-input input-text"
                  placeholder="Password"
                  data-v-1875efe0=""
                /><button
                  id="togglePassword"
                  type="button"
                  tabindex="-1"
                  class="absolute ml-3 mt-1 text-gray-500 text-lg hover:text-gray-600 focus:outline-none"
                  data-v-1875efe0=""
                >
                  <i
                    class="fa-eye absolute mt-3 ml-1 fas mr-1"
                    data-v-1875efe0=""
                  ></i></button
                ><!--v-if-->
              </div>
            </div>
          </div>
        </div>
        <div
          class="bg-gray-200 h-20 flex items-center justify-end"
          data-v-1875efe0=""
        >
          <div class="flex justify-end mr-10" data-v-1875efe0="">
            <button
              id="loginModalCancel"
              class="mr-10 btn-tertiary"
              data-v-1875efe0=""
            >
              Cancel</button
            ><button
              id="loginModalConfirm"
              class="w-24 py-2 px-2 btn btn-blue-primary"
              data-v-1875efe0=""
            >
              Login
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
